### DISCUSSION TITLE
Should we migrate to postgres instead of MariaDB ?

### DISCUSSION THREAD

**Discussion by kcramakrishna** [Digital-Assistant/digital-assistant-server#25] [General]
Keycloak strongly recommends postGres.  PostGres is well supported in AWS - RDS as well as Redshift.
The UDAN core server uses mariaDB as of now but shifting to PostGres should be easy given quarkus framework. 

As a team, we are way more comfortable with MariaDB. It may be worth our while to build competency in PostGres and move wholesale to PostGres. 

We will try Keycloak with PostGres for now and leave the core application in MariaDB. Based on our KeyCloak/Postgres learnings and comfort, we will move core application to PostGres at a later date.

Need to research this more and come to a decision.

**Comment by kcramakrishna** [comment]
Here is something from chatGPT - I initially compared Keycloak and Syncope. But towards the end, I dug deeper into Postgres vs. MariaDB. Consensus seems to be that PostGres is gaining more taction in general.

**_Especially: Keycloak seems to be optimised for PostGres:_** https://chat.openai.com/share/c77edfa0-01e6-4d6f-a482-e59ac2991128

I think we will use Keycloak with postgres for now and consider migrating out application too to postGres later. (or even Mongo depending on what makes sense)

**Reply by kcramakrishna** [reply]
@yureshwar Please use PostGres with Keycloak.

**Reply by yureshwar** [reply]
Ok sure kc

**Reply by kcramakrishna** [reply]
https://www.devtoolsacademy.com/blog/mongoDB-vs-postgreSQL

**Comment by kcramakrishna** [comment]
https://www.crunchydata.com/blog/easily-convert-json-into-columns-and-rows-with-json_table
